src/components/BookInfo/BookInfo.tsx

```diff
@@ -1,50 +1,47 @@
-import { Link, useParams } from 'react-router-dom'
-import { useGoogleBooksApi } from '../hooks/useGoogleApi'
+import { Link } from 'react-router-dom'
+import { useGetBookFromUrl } from '../../hooks/useGetBookFromUrl'
 import './BookInfo.css'
 const BookInfo = () => {
-  const param = useParams()
-
-  const [{ info, isLoading, isError }] = useGoogleBooksApi(`${param.id}`)
-
+  const { book, loading, error } = useGetBookFromUrl()
   return (
     <>
       <Link to={'/bookTask'}>Home</Link>
 
-      <h2>The Details about {info?.volumeInfo.title}</h2>
-      {isLoading && <div className="loader"></div>}
-      {isError ? (
+      <h2>The Details about {book?.title}</h2>
+      {loading && <div className="loader"></div>}
+      {error ? (
         <h2>The book id is unavailable</h2>
       ) : (
         <div className="info">
           <div className="description">
             <p>
               <strong>Book Title : </strong>
-              {info?.volumeInfo.title}
+              {book?.title}
```

Comment:
## Remember  **DRY**

Before the return you could have used destructuring to avoid repeating yourself

```tsx
const {book = {} as YOUR_BOOK_TYPE, loading, error} = useGetBookFromUrl()

const {title, subtitle, ...other} = book
```